<blockquote>
    <h1>Exercise 5.9</h1>
    <p>We will now consider the <code>Boston</code> housing data set, from the <code>MASS</code> library.</p>
    <ol>
        <li>Based on this data set, provide an estimate for the population mean of <code>medv</code>. Call this estimate $\hat{\mu}$.</li>
        <li>Provide an estimate of the standard error of $\hat{\mu}$. Interpret this result. <br>
            <i>Hint: We can compute the standard error of the sample mean by dividing the sample standard deviation by the square root of the number of observations.</i></li>
        <li>Now estimate the standard error of $\hat{\mu}$ using the bootstrap. How does this compare to your answer from 2?</li>
        <li>Based on your bootstrap estimate from 3, provide a $95 \%$ confidence interval for the mean of <code>medv</code>. Compare it to the results obtained using <code>t.test(Boston\$medv)</code>. <br>
            <i>Hint: You can approximate a $95 \%$ confidence interval using the formula $[\hat{\mu}-2SE(\hat{\mu}), \hat{\mu}+2SE(\hat{\mu})]$.</i></li>
        <li>Based on this data set, provide an estimate, $\hat{\mu}_{med}$, for the median value of <code>medv</code> in the population.</li>
        <li>We now would like to estimate the standard error of $\hat{\mu}_{med}$. Unfortunately, there is no simple formula for computing the standard error of the median. Instead, estimate the standard error of the median using the bootstrap. Comment on your findings.</li>
        <li>Based on this data set, provide an estimate for the tenth percentile of <code>medv</code> in Boston suburbs. Call this quantity $\hat{\mu}_{0.1}$. (You can use the <code>quantile()</code> function.)</li>
        <li>Use the bootstrap to estimate the standard error of $\hat{\mu}_{0.1}$. Comment on your findings.</li>
    </ol>
</blockquote>

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../../DataSets/Boston/Boston.csv")

<h3>Exercise 5.9.1</h3>
<blockquote>
    <i>Based on this data set, provide an estimate for the population mean of <code>medv</code>. Call this estimate $\hat{\mu}$.</i>
</blockquote>

In [7]:
df = df[['medv']]
mu_hat = df.mean().iloc[0]
mu_hat

22.532806324110677

<p>So $\hat{\mu} = 22.53$.</p>

<h3>Exercise 5.9.2</h3>
<blockquote>
    <i>Provide an estimate of the standard error of $\hat{\mu}$. Interpret this result. <br>
            <i>Hint: We can compute the standard error of the sample mean by dividing the sample standard deviation by the square root of the number of observations.</i>
</blockquote>

In [8]:
n = df.shape[0]
stderr_est = ((df.std(ddof=1).iloc[0])**2 / n)**0.5
stderr_est

0.40886114749753505

<p>So $\widehat{SE}(\hat{\mu}) = 0.41$.</p>

<h3>Exercise 5.9.3</h3>
<blockquote>
    <i>Provide an estimate of the standard error of $\hat{\mu}$. Interpret this result. <br>
            <i>Hint: We can compute the standard error of the sample mean by dividing the sample standard deviation by the square root of the number of observations.</i>
</blockquote>